django/db/models/sql/compiler.py

```diff
@@ -359,9 +359,14 @@ def get_order_by(self):
                     ))
             else:
                 if col not in self.query.extra_select:
-                    order_by.append((
-                        OrderBy(RawSQL(*self.query.extra[col]), descending=descending),
-                        False))
+                    if self.query.combinator:
+                        order_by.append((
+                            OrderBy(Ref(col, RawSQL(*self.query.extra[col])), descending=descending),
+                            True))
```

Comment:
Since what you're appending here is the same as what's being appended in the second `else` clause below, it looks like you could adjust your conditional to handle both of these cases, so there would be two cases in all instead of three.

Reply:
Didn't see that, good catch 👌